src/server/game/Entities/Player/Player.cpp

```diff
@@ -7309,6 +7299,31 @@ void Player::SetArenaPoints(uint32 value)
         AddKnownCurrency(ITEM_ARENA_POINTS_ID);
 }
 
+void Player::SendPVPCreditMessage(int32 honor, ObjectGuid victimGuid, uint32 victimRank)
+{
+    // Sends honor message to client (chat log, floating text)
+    
+    // Packet details:
+    //      honor - amount of honor to show in log
+    //      victim_guid - to show victim name in log
+    //      victim_rank [1..4]  HK: <dishonored rank>
+    //      victim_rank [5..19] HK: <alliance\horde rank>
+    //      victim_rank [0, 20+] HK: <>
+    //
+    //      NOTE: The "HK:" is also displayed as floating text
+
+    // Defaulting victim_guid and victim_rank to zero prevents the client from displaying an
+    //  empty "HK:" floating message if victim has no rank, such as an NPC (script: player_creature_honor)
+    //  or an invalid player (though the second case should NEVER happen)
+
+    WorldPacket data(SMSG_PVP_CREDIT, 4 + 8 + 4);
+    data << honor;
+    data << uint64((victimGuid.IsPlayer() ? victimGuid.GetCounter() : 0)); // must force uint64 or results in wrong message
```

Comment:
a full player guid in 335 branch has the same value as GetCounter (because highguid is 0) but that isnt the point, extracting uint32 counter then casting it to uint64 (packet struct requirement) because client expects a full guid doesnt make sense
This line should have been data << (victimGuid.IsPlayer() ? victimGuid : ObjectGuid::Empty)
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Object/ObjectGuid.cpp#L71 (operator << ObjectGuid)